### PR TITLE
The major refactoring after our last sync up

### PR DESCRIPTION
**Pull request by wewei**

### CODE REVIEW

File: src/utils/object.js [hunk 1]
```diff
@@ -49,3 +49,5 @@ export const isEmpty = (obj) => {
   }
   return Object.keys(obj).length === 0;
 };
+
+export const isString = obj => typeof obj === 'string';
```
Comment (BigYu) [inline src/utils/object.js]: image

Comment (BigYu) [reply src/utils/object.js]:
```javascript
Object.prototype.toString.call(obj) === '[object string]';
```

Comment (wewei) [reply src/utils/object.js]: fixed

File: src/builtin-projections/index.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
-export { defaults } from './defaults';
-export { decoration } from './decoration';
-export { columnDecoration } from './column-decoration';
-export { sortable } from './sortable';
-export { theme } from './theme';
-export { columnWidth } from './column-width';
-export { customRow } from './custom-row';
+export { default as defaults } from './defaults';
+export { default as columns } from './columns';
```
Comment (lyweiwei) [inline src/builtin-projections/index.js]: add aliast $.cols

File: src/builtin-projections/index.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+export { default as data } from './data';
```
Comment (lyweiwei) [inline src/builtin-projections/index.js]: $.data

File: src/builtin-projections/index.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+export { default as header } from './header';
```
Comment (lyweiwei) [inline src/builtin-projections/index.js]: add header row

File: src/builtin-projections/index.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
+export { default as decoration } from './decoration';
```
Comment (lyweiwei) [inline src/builtin-projections/index.js]: add $tr, $td

File: src/builtin-projections/index.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
...
+export { default as defaultContent } from './default-content';
```
Comment (lyweiwei) [inline src/builtin-projections/index.js]: Put a string as the default content.

**Pull request merged**